### PR TITLE
Btc to fiat rest api

### PR DESCRIPTION
**Pull request by itailiors**

### CODE REVIEW

**Comment by itailiors** [comment]
Generally finished,
I have these questions for this.
1) @DavidGershony we talked about that currencyservice.cs can go into balanceDisplay but the current architectural of this doesn't work. (if we look at the utxo each of them is its own component they aren't the same one).
is there a better way to design it then? (right now i can think only that it can be precomputed like it is in the PR).

2) i added `if (_NetworkConfiguration.GetNetwork().CoinTicker.Equals("BTC", StringComparison.OrdinalIgnoreCase))` because tbtc isnt real money (david's comment), so right now no users will see the actual change so we won't get feedback if theres something wrong.
should i comment it out for now?

3) there are more places I Could put balancedisplay but i'll wait to know if the logic is correct (because of 1)

**Comment by miladsoft** [comment]
> Generally finished, I have these questions for this.
> 
> 1. @DavidGershony we talked about that currencyservice.cs can go into balanceDisplay but the current architectural of this doesn't work. (if we look at the utxo each of them is its own component they aren't the same one).
>    is there a better way to design it then? (right now i can think only that it can be precomputed like it is in the PR).
> 2. i added `if (_NetworkConfiguration.GetNetwork().CoinTicker.Equals("BTC", StringComparison.OrdinalIgnoreCase))` because tbtc isnt real money (david's comment), so right now no users will see the actual change so we won't get feedback if theres something wrong.
>    should i comment it out for now?
> 3. there are more places I Could put balancedisplay but i'll wait to know if the logic is correct (because of 1)

I believe it's sufficient to display the price only in the wallet and specifically for Bitcoin. I initially planned to implement this on the main page, but considering Angor's primary mission, I see this as an unnecessary addition.

File: src/Angor/Client/Angor.Client.csproj [hunk 1]
```diff
@@ -15,7 +15,10 @@
     <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.DevServer" Version="8.0.7" PrivateAssets="all" />
     <PackageReference Include="NBitcoin" Version="7.0.25" />
     <PackageReference Include="Nostr.Client" Version="2.0.0" />
+    <PackageReference Include="System.Text.Json" Version="8.0.4" />
     <PackageReference Include="QRCoder" Version="1.6.0" />
+    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly" Version="8.0.8" />
```
Comment (dangershony) [inline src/Angor/Client/Angor.Client.csproj:20]: any idea what this references are? do we need them still?

File: src/Angor/Client/Pages/Index.razor [hunk 2]
```diff
@@ -20,10 +21,33 @@
...
             </div>
         </div>
     </div>
 </div>
 
 
+@code{
+    private string btcBalanceInUsd;
+ 
+    protected override async Task OnInitializedAsync()
```
Comment (dangershony) [inline src/Angor/Client/Pages/Index.razor:34]: so wait we fetch balance only once on startup?

File: src/Angor/Client/Services/CurrencyRateService.cs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+    public async Task<decimal> GetBtcToCurrencyRate(string currencyCode)
+    {
+        if (string.IsNullOrWhiteSpace(currencyCode))
+            throw new ArgumentException("Currency code must not be null or empty.", nameof(currencyCode));
+
+        var preferredCurrency = currencyCode.ToUpper();
+
+        // Check if the rate is cached and still valid
+        var cachedRate = GetCachedRate(preferredCurrency);
```
Comment (dangershony) [inline src/Angor/Client/Services/CurrencyRateService.cs:28]: ah I see we have caching ok

File: src/Angor/Client/Storage/LocalSessionStorage.cs [hunk 4]
```diff
@@ -42,26 +37,6 @@ public bool IsProjectMetadataStorageByPubkey(string pubkey)
         return _sessionStorageService.GetItem<Project>(projectId);
     }
 
-    public void AddProjectIdToNostrPubKeyMapping(string npub, string projectId)
```
Comment (dangershony) [inline src/Angor/Client/Storage/LocalSessionStorage.cs:45]: was this not used?

File: src/Angor/Client/Storage/LocalSessionStorage.cs [hunk 5]
```diff
@@ -96,19 +71,66 @@ public void SetUnconfirmedOutboundFunds(List<Outpoint> unconfirmedInfo)
...
     {
         _sessionStorageService.Clear();
     }
+
+    public void StoreProjectMetadata(string pubkey, ProjectMetadata projectMetadata)
+    {
+        _sessionStorageService.SetItem(pubkey, projectMetadata);
+    }
+
+    public void AddProjectIdToNostrPubKeyMapping(string npub, string projectId)
```
Comment (dangershony) [inline src/Angor/Client/Storage/LocalSessionStorage.cs:112]: ah sorry it moved down ok

File: src/Angor/Client/Pages/Wallet.razor [hunk 6]
```diff
@@ -16,23 +7,28 @@
...
 @inject IClipboardService _clipboardService
+@inject ICurrencyService _currencyService
 @inject IDerivationOperations _derivationOperations
 @inject NavMenuState NavMenuState
 @inject IEncryptionService _encryptionService
 @inject IClipboardService ClipboardService
-
-
+@using System.Text.Json
+@using Angor.Client.Storage
```
Comment (dangershony) [inline src/Angor/Client/Pages/Wallet.razor:16]: why move the `using` under the `injected` types? out convention is that they are above, please fix this.

File: src/Angor/Client/Pages/Wallet.razor [hunk 7]
```diff
@@ -1255,4 +1257,48 @@ else
...
+
+        var addresses = accountBalanceInfo.AccountInfo.AllAddresses().ToList();
+        
+        if (addresses.Count == 0)
+        {
+            Logger.LogWarning("No addresses available for processing.");
+            return;
+        }
+
+        var btcBalances = addresses
```
Comment (dangershony) [inline src/Angor/Client/Pages/Wallet.razor:1285]: why not use the already existing type `accountBalanceInfo.TotalBalance` ?

**Review by dangershony — CHANGES_REQUESTED** [review]
The tow last comments we should change before we push

**Comment by dangershony** [comment]
> considering Angor's primary mission, I see this as an unnecessary addition.

I agree, good call

File: src/Angor/Client/Pages/Index.razor (hunk 2 above)
Comment (miladsoft) [reply src/Angor/Client/Pages/Index.razor:34]: removed from index

File: src/Angor/Client/Pages/Wallet.razor (hunk 6 above)
Comment (miladsoft) [reply src/Angor/Client/Pages/Wallet.razor:16]: fixes

File: src/Angor/Client/Pages/Wallet.razor (hunk 7 above)
Comment (miladsoft) [reply src/Angor/Client/Pages/Wallet.razor:1286]: Use GetBtcValuesInPreferredCurrency for fiat conversion.

**Review by dangershony — APPROVED** [review]

File: src/Angor/Client/Angor.Client.csproj (hunk 1 above)
Comment (dangershony) [reply src/Angor/Client/Angor.Client.csproj:18]: what about this?

**Review by dangershony — APPROVED** [review]

**Pull request merged**